src/DataStream/VectorField.cc

```diff
@@ -9,162 +9,189 @@
 
 namespace carta {
 
-VectorField::VectorField() : _calculate_pi(false), _calculate_pa(false), _current_stokes_as_pi(false), _current_stokes_as_pa(false) {
-    ClearSettings();
-}
+VectorFieldCalculator::VectorFieldCalculator(const CARTA::SetVectorOverlayParameters& message, bool has_stokes_axis)
+    : _file_id(message.file_id()),
+      _smoothing_factor(message.smoothing_factor()),
+      _fractional(message.fractional()),
+      _threshold(message.threshold()),
+      _debiasing(message.debiasing()),
+      _q_error(message.debiasing() ? message.q_error() : 0),
+      _u_error(message.debiasing() ? message.u_error() : 0),
+      _stokes_intensity(message.stokes_intensity()),
+      _stokes_angle(message.stokes_angle()),
+      _compression_type(message.compression_type()),
+      _compression_quality(message.compression_quality()),
+      _threshold_option(message.threshold_option()),
+      _calculate_pi(_stokes_intensity == COMPUTED && has_stokes_axis),
+      _calculate_pa(_stokes_angle == COMPUTED && has_stokes_axis),
+      _current_stokes_as_pi((_stokes_intensity == CURRENT && has_stokes_axis) || (_stokes_intensity == COMPUTED && !has_stokes_axis)),
+      _current_stokes_as_pa((_stokes_angle == CURRENT && has_stokes_axis) || (_stokes_angle == COMPUTED && !has_stokes_axis)),
```

Comment:
These flags should now be removed; the calculator will directly query the source and threshold option values in the message and not perform any substitutions.